### ISSUE TITLE
Logging is broken

### ISSUE THREAD

**Issue by GoogleCodeExporter** [gdsaxton/python-nameparser#10]
```
In parser.py, the call to logging.basicConfig is commented out.  This prevents  
calls to log.error from working, instead displaying 'No handlers could be found 
for logger "HumanName"'.  This is easily remedied by making a call to 
logging.basicConfig(level=logging.ERROR).
```

Original issue reported on code.google.com by `jayqhac...@gmail.com` on 8 Feb 2012 at 2:49

**Comment by GoogleCodeExporter** [comment]
```
shouldnt log config be up to the caller? dont think a lib should do this.
```

Original comment by `and...@poundpay.com` on 18 Dec 2012 at 6:58

**Comment by GoogleCodeExporter** [comment]
```
The comment #1 is right: nameparser should not setup log config. This had 
actually messed up our application's logging. The "logging.basicConfig" line 
shall be removed. To receive nameparser's messages, one should add handlers 
specifically for "HumanName" logger.
```

Original comment by `alexey.s...@muranosoft.com` on 20 Dec 2012 at 10:50

**Comment by GoogleCodeExporter** [comment]
```
I'm actually not too clear on the correct solution here. I agree that the 
library shouldn't call logging.basicConfig because that will unexpectedly 
change people's logging. I believe that I need to set a handler for the logger 
to avoid the "No handlers could be found for logger" message. 

Other ideas, comment out all the logging messages or create a dummy logger 
object that does nothing for consumers of the lib but we can turn on some 
boolean flag to see logging while developing. Both of those solutions seem 
kinda stupid.

I guess I think the real problem is that python emits a warning when a logger 
does not have a handler. In this case that's exactly what I want, a logger with 
no handler and no warning about it. I can add a handler when I want to see the 
logging.

I guess I will create a StreamHandler and set it's logging level to ERROR, then 
make sure there are no log.error() calls. That should mean no logging messages 
for consumers of the lib but an easy way to change the logging level for 
development.
```

Original comment by `dere...@gmail.com` on 10 Feb 2013 at 8:36

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed in v0.2.4
```

Original comment by `dere...@gmail.com` on 11 Feb 2013 at 3:46
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
The great explanation is here: 
http://docs.python.org/2/howto/logging.html#library-config
```

Original comment by `aleksey....@gmail.com` on 11 Feb 2013 at 9:41
